### ISSUE TITLE
TextDelegateDecoder's delegation logic is incompatible with HTTP/2

### ISSUE THREAD

**Issue by jeremyk-91** [palantir/atlasdb#1607]
We've been noticing a bunch of `JsonParseException`s when reading from the endpoints that return plain-text when running with HTTP/2 (Dropwizard 1.0.6). There are two: `leader/uuid` and `timestamp-management/ping`.

In particular, it seems the header check doesn't work properly, and the Jackson decoder tries to decode output with content-type text/plain. Of course, this will fail (`pong` in text plain is read as a raw string, not `"pong"` which would be correct JSON serialization of that).

This blocks all HTTP/2 workflows; products using HTTP/2 *will not start* because the first call a client must make is a timestamp-management/ping call (as part of automated migration).

A simple workaround is to implement `TextDelegateDecoder` in a not-so-clever way:

```java
        try {
            return delegate.decode(response, type);
        } catch (Exception e) { // or JsonParseException
            return stringDecoder.decode(response, type);
        }
```

There *are* edge cases that the above won't handle correctly. Responses could for instance be the text/plain response `true` or `null`, which themselves are valid JSON.

A better long-term solution is to figure out what exactly is going on with the http2 responses, and decide how to decode responses based on that.

**Comment by hsaraogi** [comment]
Exception in thread "main" java.lang.IllegalStateException: Could not contact the Timelock Server.
	at com.palantir.atlasdb.factory.startup.TimeLockMigrator.migrate(TimeLockMigrator.java:61)
	at com.palantir.atlasdb.factory.TransactionManagers.createRawServices(TransactionManagers.java:311)
	at com.palantir.atlasdb.factory.TransactionManagers.createLockAndTimestampServices(TransactionManagers.java:285)
	at com.palantir.atlasdb.factory.TransactionManagers.create(TransactionManagers.java:154)
	at com.palantir.atlasdb.factory.TransactionManagers.create(TransactionManagers.java:128)
	at com.palantir.atlasdb.factory.TransactionManagers.create(TransactionManagers.java:115)
	at com.palantir.test.client.FoundryCatalogServer.run(FoundryCatalogServer.java:136)
	at com.palantir.test.client.FoundryCatalogServer.run(FoundryCatalogServer.java:96)
	at io.dropwizard.cli.EnvironmentCommand.run(EnvironmentCommand.java:40)
	at io.dropwizard.cli.ConfiguredCommand.run(ConfiguredCommand.java:77)
	at io.dropwizard.cli.Cli.run(Cli.java:70)
	at io.dropwizard.Application.run(Application.java:80)
	at com.palantir.test.client.FoundryCatalogServer.main(FoundryCatalogServer.java:101)
Caused by: feign.FeignException: Unrecognized token 'pong': was expecting ('true', 'false' or 'null')
 at [Source: java.io.BufferedReader@3e0fbeb5; line: 1, column: 9] reading GET https://localhost:8421/test/timestamp-management/ping
	at feign.FeignException.errorReading(FeignException.java:48)
	at feign.SynchronousMethodHandler.executeAndDecode(SynchronousMethodHandler.java:132)
	at feign.SynchronousMethodHandler.invoke(SynchronousMethodHandler.java:74)
	at feign.ReflectiveFeign$FeignInvocationHandler.invoke(ReflectiveFeign.java:97)
	at com.sun.proxy.$Proxy73.ping(Unknown Source)
	at com.palantir.atlasdb.factory.startup.TimeLockMigrator.migrate(TimeLockMigrator.java:59)
	... 12 more
Caused by: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'pong': was expecting ('true', 'false' or 'null')
 at [Source: java.io.BufferedReader@3e0fbeb5; line: 1, column: 9]
	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1581)
	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:533)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._reportInvalidToken(ReaderBasedJsonParser.java:2462)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser._handleOddValue(ReaderBasedJsonParser.java:1621)
	at com.fasterxml.jackson.core.json.ReaderBasedJsonParser.nextToken(ReaderBasedJsonParser.java:689)
	at com.fasterxml.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:3776)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3721)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2787)
	at feign.jackson.JacksonDecoder.decode(JacksonDecoder.java:65)
	at com.palantir.atlasdb.http.TextDelegateDecoder.decode(TextDelegateDecoder.java:74)
	at feign.SynchronousMethodHandler.decode(SynchronousMethodHandler.java:151)
	at feign.SynchronousMethodHandler.executeAndDecode(SynchronousMethodHandler.java:121)
	... 16 more

**Comment by hsaraogi** [comment]
The bug was detected to be root-caused to case sensitivity in the check for Content-Type Header in TextDelegateDecoder.

**Comment by jeremyk-91** [comment]
Fixed in #1613
